expertsystem/data.py

```diff
@@ -215,6 +216,37 @@ def __repr__(self) -> str:
         return f"{self.__class__.__name__}{self.name, self.pid, self.state, self.mass, self.width}"
 
 
+def compute_gellmann_nishijima(state: QuantumState) -> Optional[float]:
```

Comment:
Better change this to return a `int`

Reply:
I did that at first, but then figured this is implicit behaviour. Afaik formula itself holds just as well for `float` charges.

Reply:
But the charge from the `QuantumState` or your tests are using `int`. So then a potential cast to int will be done at that point of comparison. Hence I would rather put it in that function. It might hold for float charges, but actually measurable particle has float charge. So you could also filter this problem there. `if (charge).is_integer(): return int(charge)`

Reply:
I think the filtering should take place here:
https://github.com/ComPWA/expertsystem/blob/c0abac05fc3d50bb638f079dc77bd600e1d3e227/expertsystem/data.py#L192-L195
This check will fail if the GM fomula returned a `float` that is not an integer.

Reply:
> `if (charge).is_integer(): return int(charge)`

To be clear: this would also mean raising an exception if the charge is not castable to `int`. For me that is unexpected behaviour in a function that is just supposed to compute the Gell-Mann–Nishijima formula (as its name states).

Reply:
But by definition this should always be an integer. So if that happens to be not an integer you can raise and exception at the point of cause. That would make more sense to me

Reply:
Okay, let me loosen my reasoning here and go the physics route in an attempt to prove that the GM formula can only concern integers (I haven't found this as an explicit statement in any papers yet).
The GM formula relates baryon number, strangeness etc. The fact that we talk about baryon number means that we cannot have isolated quarks (right, or is there some NP that I'm overlooking here?). That means that the `QuantumState` we are testing here can only have whole charge.
It still seems like a lot of implicit assumptions for a function that should just compute a formula, but I'll put that intelligence in there.

Reply:
Hmm, you have a point. So I think you are right, we do not have to put any more assumptions in there. I just want to make sure we are actually catching out the ill defined particles.

Reply:
I think because `particle.charge` is of type `int`, we are on the safe side.

Reply:
`Particle` instances with `float` charge raise an exception, even if they comply with the Gell-Mann–Nishijima formula. `QuantumState` instances are allowed, but there's no check on GM formula there anyway.

Reply:
And mypy would fish out float charges anyways right?

Reply:
Yes, but mypy isn't checked on outside scripts, it's only used as a consistency check for the framework and tests themselves. But for a user running, say, a Jupyter notebook, an exception would be raised for any `Particle` that has unphysical charge (as per Gell-Mann–Nishijima formula + cast check). `QuantumState` remains an internal matter, however.